main/main.go

```diff
@@ -0,0 +1,128 @@
+package main
```

Comment:
Seems like golang's `testing` package [already includes benchmarking capabilities](https://golang.org/pkg/testing/#hdr-Benchmarks) (you can run benchmarks separately with `go test -bench .`)